core/trino-main/src/main/java/io/trino/sql/planner/NullabilityAnalyzer.java

```diff
@@ -65,7 +65,7 @@ protected Void visitCast(Cast node, AtomicBoolean result)
             // except for the CAST(NULL AS x) case -- we should fix this at some point)
             //
             // Also, try_cast (i.e., safe cast) can return null
-            result.set(node.isSafe() || !node.isTypeOnly());
+            result.compareAndSet(false, node.isSafe() || !node.isTypeOnly());
```

Comment:
` result.set(result.get && ...)` would be clearer, since we're not using any concurrency here